### PR TITLE
add eye op, kernel and unitest test=develop

### PR DESCRIPTION
**Pull request by ForFishes**
Add eye op implementation and the unit test codes of it. This op constructs an identity matrix or a batch of matrices.

### CODE REVIEW

**Comment by ForFishes** [comment]
MacHi 2019-08-02 15-44-48

File: python/paddle/fluid/layers/tensor.py [hunk 1]
```diff
@@ -991,6 +991,77 @@ def diag(diagonal):
...
+    Examples:
+        .. code-block:: python
+
+          import paddle.fluid as fluid
+ 	  data = fluid.layers.eye(3, dtype='int32')
+	  # [[1, 0, 0]
+          #  [0, 1, 0]
+	  #  [0, 0, 1]]
+    
+          data = fluid.layers.eye(3,2, dtype='int32')
```
Comment (hutuxian) [inline python/paddle/fluid/layers/tensor.py]: Output shape is not appropriate

Comment (ForFishes) [reply python/paddle/fluid/layers/tensor.py]: done

**Comment by ForFishes** [comment]
MacHi 2019-08-06 17-03-03

**Review by hutuxian — APPROVED** [review]
LGTM for document part.

**Comment by gavin1332** [comment]
LGTM

**Review by gavin1332 — APPROVED** [review]

**Review by frankwhzhang — APPROVED** [review]

**Pull request merged**